crates/bevy_reflect/derive/src/impls/structs.rs

```diff
@@ -129,6 +129,11 @@ pub(crate) fn impl_struct(reflect_struct: &ReflectStruct) -> proc_macro2::TokenS
                 #(dynamic.insert_boxed(#field_names, #bevy_reflect_path::Reflect::clone_value(&self.#field_idents));)*
                 dynamic
             }
+
+            fn get_represented_kind_info(&self) -> #FQOption<&#bevy_reflect_path::StructInfo> {
+                let type_info = #bevy_reflect_path::Reflect::get_represented_type_info(self)?;
+                type_info.as_struct().ok()
+            }
```

Comment:
Alternatively we can probably have the trait supply a default implementation. I’m not sure which is better or if the compiler prefers one over the other for optimization purposes.

But putting it in a default method at least reduces the amount of generated code.

Reply:
Went for the default impl